Set/Models/GameModel.swift

```diff
@@ -122,15 +126,37 @@ struct GameModel {
         }
     }
     
-    mutating private func getBonus(for player: Player) {
+    private mutating func isFasterThanPreviousMatch() -> Bool {
+        let matchDate = Date.now
+        let matchTimeInterval = matchDate.timeIntervalSince(previousMatchDate ?? startGameDate)
+        
+        guard let previousMatchTimeInterval = previousMatchTimeInterval else {
+            self.previousMatchTimeInterval = matchTimeInterval
+            previousMatchDate = matchDate
+            return false
+        }
+        
+        self.previousMatchTimeInterval = matchTimeInterval
+        previousMatchDate = matchDate
+        
+        return matchTimeInterval < previousMatchTimeInterval
+    }
+    
+    private mutating func getBonus(for player: Player) {
```

Comment:
Please pay attention on function naming `getBonus` but it doesn't return anything. It can mean only one - something strange is happening here..
If you name it `getBonus` it should return bonus, 0 or more,
here is again issue about single responsibility